encoder/csv.go

```diff
@@ -168,7 +130,8 @@ func NewCSVItemIterator(sourceConfig map[string]any, data io.ReadCloser) (*CSVIt
 
 	columnSeparator, ok := sourceConfig["columnSeparator"]
 	if ok {
-		reader.separator = columnSeparator.(string)
+		// how to make this a rune in best way possible '' <- seems to be a way to say it's a rune, can we do that on a string?
+		reader.decoder.Comma = rune(columnSeparator.(string)[0])
```

Comment:
Not clear how this would work for \t etc. I suggest there could be a function getSeperatorRune that takes a string and pattern matches / switch it and then returns the correct rune. Something like this:

func stringToRune(input string) (rune, error) {
    switch input {
    case ",":
        return ',', nil
    case "\t":
        return '\t', nil
    case " ":
        return ' ', nil
    default:
        return 0, errors.New("input string does not match allowed characters")
    }
}